src/test/java/api/InstanceRelationshipsTest.java

```diff
@@ -136,6 +141,299 @@ public void canFetchMultipleInstancesWithAllRelationshipsTypes() throws Exceptio
+    assertThat(response.getContentType(), is("application/json"));
+    assertThat(response.getJson(), is(expectedErrorResponse));
+  }
+
+  @Test
+  public void cannotCreateAnInstanceWithNonExistedParentInstanceId()
+    throws InterruptedException, MalformedURLException, TimeoutException,
+    ExecutionException {
+
+    String superInstanceId = UUID.randomUUID().toString();
+    final JsonObject expectedErrorResponse = new JsonObject()
+      .put("message", String.format("Cannot set instance_relationship.superInstanceId = " +
+        "%s because it does not exist in instance.id.", superInstanceId));
+    instanceRelationshipClient.emulateFailure(
+      new EndpointFailureDescriptor()
+        .setFailureExpireDate(DateTime.now().plusSeconds(2).toDate())
+        .setStatusCode(422)
+        .setContentType("application/json")
+        .setBody(expectedErrorResponse.toString())
+        .setMethod(HttpMethod.POST));
+
+    JsonObject parentInstance = new JsonObject()
+      .put("id", UUID.randomUUID().toString())
+      .put("superInstanceId", superInstanceId);
+
+    JsonArray parentInstances = new JsonArray().add(parentInstance);
+
+    JsonObject smallAngryPlanetJson = smallAngryPlanet(UUID.randomUUID());
+    smallAngryPlanetJson.put("parentInstances", parentInstances);
+
+    Response response = instancesClient.attemptToCreate(smallAngryPlanetJson);
+
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getContentType(), is("application/json"));
+    assertThat(response.getJson(), is(expectedErrorResponse));
+  }
+
+  @Test
+  public void cannotCreateAnInstanceWithNonExistedChildInstanceId()
+    throws InterruptedException, MalformedURLException, TimeoutException,
+    ExecutionException {
+
+    String subInstanceId = UUID.randomUUID().toString();
+    final JsonObject expectedErrorResponse = new JsonObject()
+      .put("message", String.format("Cannot set instance_relationship.subInstanceId = " +
+        "%s because it does not exist in instance.id.", subInstanceId));
+    instanceRelationshipClient.emulateFailure(
+      new EndpointFailureDescriptor()
+        .setFailureExpireDate(DateTime.now().plusSeconds(2).toDate())
+        .setStatusCode(422)
+        .setContentType("application/json")
+        .setBody(expectedErrorResponse.toString())
+        .setMethod(HttpMethod.POST));
+
+    JsonObject childInstance = new JsonObject()
+      .put("id", UUID.randomUUID().toString())
+      .put("subInstanceId", subInstanceId);
+
+    JsonArray childInstances = new JsonArray().add(childInstance);
+
+    JsonObject smallAngryPlanetJson = smallAngryPlanet(UUID.randomUUID());
+    smallAngryPlanetJson.put("childInstances", childInstances);
+
+    Response response = instancesClient.attemptToCreate(smallAngryPlanetJson);
+
+    assertThat(response.getStatusCode(), is(422));
+    assertThat(response.getContentType(), is("application/json"));
+    assertThat(response.getJson(), is(expectedErrorResponse));
+  }
+
```

Comment:
Could you please add a test that verifies that we can not create an instance with non existed relationship type id?